rosgraph_monitor/src/monitor.cpp

```diff
@@ -668,14 +696,44 @@ void RosGraphMonitor::fill_rosgraph_msg(rosgraph_monitor_msgs::msg::Graph & msg)
         node_msg.subscriptions.push_back(topic_msg);
       }
     }
-
     msg.nodes.push_back(node_msg);
   }
 }
 
-void RosGraphMonitor::set_graph_change_callback(std::function<void()> callback)
+void RosGraphMonitor::set_graph_change_callback(
+  std::function<void(rosgraph_monitor_msgs::msg::Graph &)> callback)
+{
+  graph_change_callback_ = [callback, this]() {
+      rosgraph_monitor_msgs::msg::Graph msg;
+      fill_rosgraph_msg(msg);
+      callback(msg);
+    };
+}
+
+void RosGraphMonitor::query_node_parameters(const std::string & node_name)
 {
-  graph_change_callback_ = callback;
+  // Non-blocking async call for parameter query. Hold onto the future to track completion.
+  params_futures[node_name] = query_params_(
+    node_name,
+    [this, node_name_copy = std::string(node_name)](
+      const rcl_interfaces::msg::ListParametersResult & result) {
+      RCLCPP_INFO(
+        logger_, "Got parameters for node %s: %zu", node_name_copy.c_str(),
+        result.names.size());
+      auto & tracking = nodes_[node_name_copy];
+      for (const auto & param_name : result.names) {
+        tracking.params.push_back(
+          ParameterTracking{param_name,
+            rcl_interfaces::msg::ParameterType::PARAMETER_NOT_SET});
+      }
+      if (!tracking.params.empty()) {
+        // Although the querying of node parameters doesn't necessarily
```

Comment:
Valid but a little tautological: "Understanding of graph changed" is always the triggering event :)